File: app/src/main/java/me/devsaki/hentoid/timber/CrashlyticsTree.java
package me.devsaki.hentoid.timber;

import androidx.annotation.NonNull;
import android.util.Log;

import com.crashlytics.android.Crashlytics;

import timber.log.Timber;

public class CrashlyticsTree extends Timber.Tree {

    @Override
    protected boolean isLoggable(String tag, int priority) {
        return priority >= Log.WARN;
    }

    @Override
    protected void log(int priority, String tag, @NonNull String message, Throwable t) {
        Crashlytics.log(message);
        if (t != null) Crashlytics.logException(t);
    }
}

File: app/src/main/java/me/devsaki/hentoid/timber/CrashlyticsTree.kt
package me.devsaki.hentoid.timber

import android.util.Log
import com.google.firebase.crashlytics.FirebaseCrashlytics
import timber.log.Timber


class CrashlyticsTree : Timber.Tree() {

    override fun isLoggable(tag: String?, priority: Int) = priority >= Log.WARN

    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
        val crashlytics = FirebaseCrashlytics.getInstance()
        if (t != null) crashlytics.recordException(t)
        crashlytics.log(message)
    }
}
